core/src/main/scala/kafka/log/Log.scala

```diff
@@ -464,6 +466,8 @@ class Log(@volatile var dir: File,
   }
 
   private def loadProducerState(lastOffset: Long, reloadFromCleanShutdown: Boolean): Unit = lock synchronized {
+    if (dir == null)
+      throw new KafkaStorageException(s"The log for partition $topicPartition is offline")
```

Comment:
Could we put the above two lines in a private checkIfLogOffline() method and reuse?

Reply:
Good point. I will update the patch as suggested.